### PR TITLE
Fix:(issue_1293) Wrap usage text for commands

### LINKED ISSUE #1293: Usage does not auto indent newlines
## my urfave/cli version is

```
v2.3.0
```

## Checklist

* [x] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [x] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/master/docs/v2/manual.md)
* [x] Did you perform a search about this problem? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## Dependency Management

- [x] My project is using go modules.
- [ ] My project is using vendoring.
- [ ] My project is automatically downloading the latest version.
- [ ] I am unsure of what my dependency management setup is.

## Describe the bug

Usage does not auto indent newlines in usage.

Expected output:

```
$ ./foo 
NAME:
   foo - sample usage

USAGE:
   foo [global options] command [command options] [arguments...]

COMMANDS:
   run  this is a long help output for the run command, long usage
        output, long usage output, long usage output, long usage output  <-- notice the indent
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --help, -h  show help (default: false)
```

Actual output:

```
$ ./foo 
NAME:
   foo - sample usage

USAGE:
   foo [global options] command [command options] [arguments...]

COMMANDS:
   run  this is a long help output for the run command, long usage
output, long usage output, long usage output, long usage output  <-- not auto indenting
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --help, -h  show help (default: false)
```

## To reproduce

<details>
<summary>Example</summary>
<br>

```go
package main

import (
	"fmt"
	"os"

	"github.com/urfave/cli/v2"
)

func main() {
	app := &cli.App{
		Usage: "sample usage",
		Commands: []*cli.Command{
			{
				Name:  "run",
				Usage: "this is a long help output for the run command, long usage\noutput, long usage output, long usage output, long usage output",
				Action: func(c *cli.Context) error {
					return fmt.Errorf("nothing")
				},
			},
		},
	}

	err := app.Run(os.Args)
	if err != nil {
		fmt.Fprintf(os.Stdout, "%s: %s\n", os.Args[0], err)
		os.Exit(1)
	}
}
```
</details>

## Additional context

Related to: #1278

## Run `go version` and paste its output here

```
go version go1.16.5 linux/amd64
```

## Run `go env` and paste its output here

```
not relevant 
```

### PR DESCRIPTION
**Pull request by dearchap**
## What type of PR is this?

_(REQUIRED)_

- bug

## What this PR does / why we need it:

_(REQUIRED)_

Fix command usage wrap for multiline text
## Which issue(s) this PR fixes:

_(REQUIRED)_

## Special notes for your reviewer:

_(fill-in or delete this section)_

## Testing

_(fill-in or delete this section)_

go test -run=TestShowAppHelp_CommandMultiLine_UsageText

## Release Notes

_(REQUIRED)_

```release-note

```

### CODE REVIEW

**Comment by WestleyK** [comment]
Thanks for looking into this, so heres the output with your changes here:

```
$ ./foo 
NAME:
   foo - sample usage

USAGE:
   foo [global options] command [command options] [arguments...]

COMMANDS:
   test  long help for test command
   foo   long help for foo command
        this command does nothing.
   run  this is a long help output for the run command, long usage
        output, long usage output, long usage output, long usage output
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --help, -h  show help (default: false)
```

Better, but the indenting is not consistent with the other commands, heres what I would expect it to look like:

```
$ ./foo 
NAME:
   foo - sample usage

USAGE:
   foo [global options] command [command options] [arguments...]

COMMANDS:
   test     long help for test command
   foo      long help for foo command
            this command does nothing.
   run      this is a long help output for the run command, long usage
            output, long usage output, long usage output, long usage output
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --help, -h  show help (default: false)
```

Heres what it looks like without any newlines (before your changes):

```
$ ./foo 
NAME:
   foo - sample usage

USAGE:
   foo [global options] command [command options] [arguments...]

COMMANDS:
   test     long help for test command
   foo      long help for foo command this command does nothing.
   run      this is a long help output for the run command, long usage output, long usage output, long usage output, long usage output
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --help, -h  show help (default: false)
```

File: help_test.go [hunk 1]
```diff
@@ -895,6 +895,35 @@ App UsageText`,
...
+	}
+
+	output := &bytes.Buffer{}
+	app.Writer = output
+
+	_ = app.Run([]string{"foo"})
+
+	expected := `frobbly, frb1, frbb2, frl2  this is a long help output for the run command, long usage 
+                               output, long usage output, long usage output, long usage output
+                               output, long usage output, long usage output`
```
Comment (mostynb) [inline help_test.go:920]: It might be worth specifying the entire COMMANDS block as the expected input, so we can see how the indentation works.

**Comment by dearchap** [comment]
@WestleyK Give it a shot now.

File: help_test.go (hunk 1 above)
Comment (dearchap) [reply help_test.go:920]: Let me try. It is very difficult to get the spacing/tabs exactly right

**Comment by WestleyK** [comment]
Looks perfect! Thanks for taking care of this issue!

```
$ ./foo 
NAME:
   foo - sample usage

USAGE:
   foo [global options] command [command options] [arguments...]

COMMANDS:
   test     long help for test command
   foo      long help for foo command
            this command does nothing.
   run      this is a long help output for the run command, long usage
            output, long usage output, long usage output, long usage output
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --help, -h  show help (default: false)
```

<details><summary>Code used</summary>
<p>

```go
package main

import (
	"fmt"
	"os"

	"github.com/urfave/cli/v2"
)

func main() {
	app := &cli.App{
		Usage: "sample usage",
		Commands: []*cli.Command{
			{
				Name:  "test",
				Usage: "long help for test command",
			},
			{
				Name:  "foo",
				Usage: "long help for foo command\nthis command does nothing.",
			},
			{
				Name:  "run",
				Usage: "this is a long help output for the run command, long usage\noutput, long usage output, long usage output, long usage output",
				Action: func(c *cli.Context) error {
					return fmt.Errorf("nothing")
				},
			},
		},
	}

	err := app.Run(os.Args)
	if err != nil {
		fmt.Fprintf(os.Stdout, "%s: %s\n", os.Args[0], err)
		os.Exit(1)
	}
}
```

go.mod
```go
module foo

go 1.18

require github.com/urfave/cli/v2 v2.11.2

require (
	github.com/cpuguy83/go-md2man/v2 v2.0.2 // indirect
	github.com/russross/blackfriday/v2 v2.1.0 // indirect
	github.com/xrash/smetrics v0.0.0-20201216005158-039620a65673 // indirect
)

replace github.com/urfave/cli/v2 => github.com/dearchap/cli/v2 v2.3.1-0.20220824131205-915fd7adcaa5
```

</p>
</details>

Should also update the documentation that `Usage` allows newlines.

File: help_test.go (hunk 1 above)
Comment (dearchap) [reply help_test.go]: Updated the test case

**Comment by dearchap** [comment]
@WestleyK would you like to submit a PR for the documentation ?

File: help_test.go [hunk 2]
```diff
@@ -895,6 +895,42 @@ App UsageText`,
...
+	app.Writer = output
+
+	_ = app.Run([]string{"foo"})
+
+	expected := "COMMANDS:\n" +
+		"   frobbly, frb1, frbb2, frl2  this is a long help output for the run command, long usage \n" +
+		"                               output, long usage output, long usage output, long usage output\n" +
+		"                               output, long usage output, long usage output\n" +
+		"   grobbly, grb1, grbb2        this is another long help output for the run command, long usage \n" +
+		"                               output, long usage output"
```
Comment (mostynb) [inline help_test.go:928]: This is nice and readable- thanks!

File: help.go [hunk 3]
```diff
@@ -476,3 +482,14 @@ func wrapLine(input string, offset int, wrapAt int, padding string) string {
 func offset(input string, fixed int) int {
 	return len(input) + fixed
 }
+
```
Comment (mostynb) [inline help.go:485]: I think this function deserves a comment.

**Review by mostynb — DISMISSED** [review]
lgtm with one small documentation request

File: help.go (hunk 3 above)
Comment (dearchap) [reply help.go:485]: Yes.

**Review by mostynb — APPROVED** [review]

**Pull request merged**